addons/mail/static/src/components/attachment/attachment.js

```diff
@@ -119,8 +133,19 @@ class Attachment extends Component {
      */
     _onClickUnlink(ev) {
         ev.stopPropagation();
-        this.attachment.remove();
-        this.trigger('o-attachment-removed');
+        if (this.attachment.isLinkedToComposer) {
+            this._unlinkAttachment();
+        } else {
+            Dialog.confirm(this,
```

Comment:
It's not exactly that, there is also the case of removal in attachment box.
Composer case really is the particular case as it's the only place where we don't ask for confirmation

Reply:
I feel that having a prompt confirmation step should not be default behaviour (because doing so encourages having a bad UX with many dialog prompts everywhere). So when you have one, it should be clear why you want it: the flows that are affected must be evident just at looking at the conditions.

Reply:
I understand but I don't think it's possible for attachment box, the only difference I see between an attachment uploaded directly through attachment box and an attachment in composer is the link with a composer